src/datatypes/compiler-minecraft.js

```diff
@@ -61,6 +62,56 @@ module.exports = {
       code += '}\n'
       code += 'return { value: data, size }'
       return compiler.wrapCode(code)
+    }],
+    bitflags: ['parametrizable', (compiler, { type, flags, shift, big }) => {
+      let fstr = JSON.stringify(flags)
+      if (Array.isArray(flags)) {
+        fstr = '{'
+        for (const [k, v] of Object.entries(flags)) fstr += `"${v}": ${big ? (1n << BigInt(k)) : (1 << k)}` + (big ? 'n,' : ',')
+        fstr += '}'
+      } else if (shift) {
+        fstr = '{'
+        for (const key in flags) fstr += `"${key}": ${1 << flags[key]},`
+        fstr += '}'
+      }
+      return compiler.wrapCode(`
+        const { value: _value, size } = ${compiler.callType(type, 'offset')}
+        const value = { _value }
+        const flags = ${fstr}
+        for (const key in flags) {
+          value[key] = (_value & flags[key]) == flags[key]
+        }
+        return { value, size }
+      `.trim())
+    }],
+    registryEntryHolder: ['parametrizable', (compiler, opts) => {
+      return compiler.wrapCode(`
+const { value: n, size: nSize } = ${compiler.callType('varint')}
+if (n !== 0) {
+  return { value: { ${opts.baseName}: n - 1 }, size: nSize }
+} else {
+  const holder = ${compiler.callType(opts.otherwise.type)}
+  return { value: { ${opts.otherwise.name}: holder.data }, size: nSize + holder.size }
+}
+      `.trim())
+    }],
+    registryEntryHolderSet: ['parametrizable', (compiler, opts) => {
+      return compiler.wrapCode(`
+  const { value: n, size: nSize } = ${compiler.callType('varint')}
+  if (n === 0) {
+    const base = ${compiler.callType(opts.base.type)}
+    return { value: { ${opts.base.name}: base.value }, size: base.size + nSize }
+  } else {
+    const set = []
+    let accSize = nSize
+    for (let i = 0; i < n - 1; i++) {
+      const entry = ${compiler.callType(opts.otherwise.type)}
```

Comment:
change to `const entry = ${compiler.callType(opts.otherwise.type, 'offset + accSize')}`

Reply:
Done